app/src/main/java/pm/gnosis/heimdall/data/repositories/impls/WalletConnectBridgeRepository.kt

```diff
@@ -224,21 +226,29 @@ class WalletConnectBridgeRepository @Inject constructor(
         }
 
     private fun showSendTransactionNotification(peerMeta: Session.PeerMeta?, safe: Solidity.Address, data: TransactionData, referenceId: Long) {
-        val icon = peerMeta?.icons?.firstOrNull()?.let { nullOnThrow { picasso.load(it).get() } }
+        val keyguard = context.getSystemService(Context.KEYGUARD_SERVICE) as KeyguardManager
         val intent = ReviewTransactionActivity.createIntent(context, safe, data, referenceId)
-        val notification = localNotificationManager.builder(
-            peerMeta?.name ?: context.getString(R.string.unknown_dapp),
-            context.getString(R.string.notification_new_transaction_request),
-            PendingIntent.getActivity(context, 0, intent, PendingIntent.FLAG_UPDATE_CURRENT),
-            CHANNEL_WALLET_CONNECT_REQUESTS
-        )
-            .setSubText(safe.shortChecksumString())
-            .setLargeIcon(icon)
-            .build()
-        localNotificationManager.show(
-            referenceId.hashCode(),
-            notification
-        )
+        // Pre Android Q we will directly show the review activity if the phone is unlocked, else we show a notification
+        // TODO: Adjust check when Q is released
+        if (BuildCompat.isAtLeastQ() || Build.VERSION.SDK_INT > Build.VERSION_CODES.P || keyguard.isKeyguardLocked) {
+            val icon = peerMeta?.icons?.firstOrNull()?.let { nullOnThrow { picasso.load(it).get() } }
+            val notification = localNotificationManager.builder(
+                peerMeta?.name ?: context.getString(R.string.unknown_dapp),
+                context.getString(R.string.notification_new_transaction_request),
+                PendingIntent.getActivity(context, 0, intent, PendingIntent.FLAG_UPDATE_CURRENT),
+                CHANNEL_WALLET_CONNECT_REQUESTS
+            )
+                .setSubText(safe.shortChecksumString())
+                .setLargeIcon(icon)
+                .build()
+            localNotificationManager.show(
+                referenceId.hashCode(),
+                notification
+            )
+        } else {
+            intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
```

Comment:
should we add this flag to createIntent of ReviewTransactionActivity?

Reply:
I only add this flag, because else it will not work (context is a service so the flag is required). Normally I would prefer not to have this flag here at all.